otherlibs/unix/unixsupport.c

```diff
@@ -288,18 +286,17 @@ int code_of_unix_error (value error)
 void unix_error(int errcode, const char *cmdname, value cmdarg)
 {
   value res;
+  const value * unix_error_exn;
   value name = Val_unit, err = Val_unit, arg = Val_unit;
 
   Begin_roots3 (name, err, arg);
     arg = cmdarg == Nothing ? caml_copy_string("") : cmdarg;
     name = caml_copy_string(cmdname);
     err = unix_error_of_code (errcode);
-    if (unix_error_exn == NULL) {
-      unix_error_exn = caml_named_value("Unix.Unix_error");
-      if (unix_error_exn == NULL)
-        caml_invalid_argument("Exception Unix.Unix_error not initialized,"
-                         " please link unix.cma");
-    }
+    unix_error_exn = caml_named_value("Unix.Unix_error");
+    if (unix_error_exn == NULL)
+      caml_invalid_argument("Exception Unix.Unix_error not initialized,"
+                       " please link unix.cma");
```

Comment:
That's OK for the time being.  Later, either check that `caml_named_value` is fast enough (currently it uses a rather primitive hash table implementation) or consider caching the result in thread-local storage or in global storage but with atomic update.